scripts/lib/CIME/hist_utils.py

```diff
@@ -343,6 +343,10 @@ def cprnc(model, file1, file2, case, rundir, multiinst_driver_compare=False, out
     if outfile_suffix:
         output_filename += ".{}".format(outfile_suffix)
 
+    # Remove existing output file if it exists
+    if os.path.exists(output_filename):
+        os.remove(output_filename)
+
```

Comment:
My read of the logic in this function is that this removal should only be done in the `else` block a few lines down – otherwise, nothing is written to `output_filename`, and in fact `output_filename` may not be writeable (and thus may not be removable) due to permission issues. Do you agree?